### PR TITLE
BME680 module initial commit

### LINKED ISSUE #2114: Module for BME680
### Missing feature
Module for BME680 is missing

### Justification
The new [BME680 sensor ](https://www.bosch-sensortec.com/bst/products/all_products/bme680)from bosch is the follow-up chip of the already supportet BME280.
Beside temperature, humidity and air pressure, **AIR QUALITY** (VOCs) could be messured.
In my opinion the BME680 is a cheap solution to messure IAQ-Index (Indoor Air Quality) for IoT and home automatisation solutions like automatic aeration.

### Workarounds
The only workaround is to directly communicate via I2C to the Chip.

### PR DESCRIPTION
**Pull request by vsky279**
Fixes #2114.

Make sure all boxes are checked (add x inside the brackets) when you submit your contribution, remove this sentence before doing so.

- [x] This PR is for the `dev` branch rather than for `master`.
- [x] This PR is compliant with the [other contributing guidelines](https://github.com/nodemcu/nodemcu-firmware/blob/dev/CONTRIBUTING.md) as well (if not, please describe why).
- [x] I have thoroughly tested my contribution.
- [x] The code changes are reflected in the documentation at `docs/en/*`.

This is a simple driver of the Bosch's BME680 sensor. In addition to the BME280 it is able to measure the indoor air quality. Unfortunately the IAQ (Indoor Air Quality) Index calculation algorithm is not publicly available. So we need to live with the gas resistance value which should still be some proxy to the IAQ.

Definitely it would be good if someone can further test the module before it is merged.

### CODE REVIEW

File: docs/en/modules/bme680.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
+# BME680 module
+| Since  | Origin / Contributor  | Maintainer  | Source  |
+| :----- | :-------------------- | :---------- | :------ |
+| 2017-10-28 | [vsky279](https://github.com/vsky279) | [vsky279](https://github.com/vsky279) | [bme680.c](../../../app/modules/bme680.c)|
+
+Copyright (C) 2017 - 2018 Bosch Sensortec GmbH
```
Comment (marcelstoer) [inline docs/en/modules/bme680.md]: What is this copyright for?

**Comment by marcelstoer** [comment]
>  it would be good if someone can further test the module

I'm afraid that due to lack of hardware this will probably not happen. However, a code review can still be conducted.

File: docs/en/modules/bme680.md (hunk 1 above)
Comment (vsky279) [reply docs/en/modules/bme680.md]: I'm using parts of the [bme680.c](https://github.com/BoschSensortec/BME680_driver/blob/2a51b9c0c1899f28e561e6701caa22cb23201cfc/bme680.c) driver from Bosch - it says we should retain the copyright. I think maybe just one line should be enough.

Comment (marcelstoer) [reply docs/en/modules/bme680.md]: I saw that, yes. However, I thought leaving the copyright in the source code would be enough. It feels somewhat out of place here and makes it look like Bosch contributed the NodeMCU module (and wrote this documentation).

Comment (vsky279) [reply docs/en/modules/bme680.md]: This seems reasonable. I will make a block in the code with these functions and will put the (c) note there.

**Comment by pjsg** [comment]
I see that this uses doubles to perform calculations... Does this work in an integer build?

**Comment by vsky279** [comment]
Yes, it does. This is even tested in the BME280 module.

**Comment by KT819GM** [comment]
Temperature readings looks wrong (upper side by about 10C). Any advice how to add compensation for this? Humidity and resistance calculations seems OK. 
Before have tried this module with ESP32, comparing readings with BME280 and had less than degree difference. 
Using sample code from sample code with: Heater temp 360C, heater time 200ms, readout delay 250ms. Wrong temperature reading renders dew calculations wrong.

**Pull request merged**